bigtable/integration_test.go

```diff
@@ -1450,17 +1380,22 @@ func TestIntegration_FullReadStats(t *testing.T) {
 						t.Errorf("CellsReturnedCount did not match. got: %d, want: %d",
 							readStats.CellsReturnedCount, test.cellsReturnedCount)
 					}
-					if readStats.CellsSeenCount != test.cellsSeenCount {
-						t.Errorf("CellsSeenCount did not match. got: %d, want: %d",
-							readStats.CellsSeenCount, test.cellsSeenCount)
-					}
 					if readStats.RowsReturnedCount != test.rowsReturnedCount {
 						t.Errorf("RowsReturnedCount did not match. got: %d, want: %d",
 							readStats.RowsReturnedCount, test.rowsReturnedCount)
 					}
-					if readStats.RowsSeenCount != test.rowsSeenCount {
-						t.Errorf("RowsSeenCount did not match. got: %d, want: %d",
-							readStats.RowsSeenCount, test.rowsSeenCount)
+					// We use lenient checks for CellsSeenCount and RowsSeenCount. Exact checks would be brittle.
+					// Note that the emulator and prod sometimes yield different values:
+					// - Sometimes prod scans fewer cells due to optimizations that allow prod to skip cells.
+					// - Sometimes prod scans more cells due to to filters that must rescan cells.
+					// Similar issues apply for RowsSeenCount.
+					if readStats.CellsSeenCount < readStats.CellsReturnedCount {
```

Comment:
Nit. I believe golang style is to use got, want to make it easier to read the if statement:

‘’’
if got, want := readStats.CellsSeenCount, readStats.CellsReyurnedCount; got < want {
…
}
‘’’

Reply:
Fixed.